src/App.tsx

```diff
@@ -34,28 +34,22 @@ import useInterval from './hooks/useInterval';
 const DashboardSidebar = () => {
   const config = useConfig();
   const astarte = useAstarte();
-  const appEngineHealth = useFetch(astarte.client.getAppengineHealth);
-  const realmManagementHealth = useFetch(astarte.client.getRealmManagementHealth);
-  const pairingHealth = useFetch(astarte.client.getPairingHealth);
-  const flowHealth = config.features.flow ? useFetch(astarte.client.getFlowHealth) : null;
-
-  const refreshHealth = useCallback(() => {
-    if (!astarte.isAuthenticated) {
-      return;
-    }
-    appEngineHealth.refresh();
-    realmManagementHealth.refresh();
-    pairingHealth.refresh();
-    if (flowHealth) {
-      flowHealth.refresh();
+
+  const healthFetcher = useFetch(() => {
```

Comment:
having a single fetch will not display results untill ALL promises are resolved.
If one service is not available and the request timesout, you won't see the result of healthy services untill the timeout kicks in.
No sure if that's a good choice for the home page service table.
It will be ok for the navbar though

Reply:
I agree, that's fine for the navbar (Promise.all rejects as soon as one of its promises rejects), but not for the homepage. However there are no warnings for hooks there and the code is already fine.

Reply:
Warnings won't prevent the build now.
I'd rather take my time to fix this properly rather than lowering the user experience for the user

Reply:
I'd say the code looks fine. Also, we will probably rework it again when we start using Redux for state and async data. So here I'd rather keep the simplest change that does the job